### PR TITLE
[mounts] More common mount plumbing refactoring

### PR DESCRIPTION
**Pull request by townsend2010**

### CODE REVIEW

File: src/client/cli/cmd/mount.cpp [hunk 1]
```diff
@@ -90,6 +90,22 @@ mp::ReturnCode cmd::Mount::run(mp::ArgParser* parser)
...
+            const auto [username, password] = mcp::get_user_password(term);
+
+            request.mutable_user_credentials()->set_username(username);
+            request.mutable_user_credentials()->set_password(password);
+
+            client->Write(request);
+
+            return;
+        }
+
```
Comment (luis4a0) [inline src/client/cli/cmd/mount.cpp:108]: If I am correct, this code is repeated verbatim in `restart.cpp` and `start.cpp`. Could this fragment be refactored into a new function?

**Review by luis4a0 — CHANGES_REQUESTED** [review]
Hey @townsend2010, thanks for this! Only two issues I see. The first one is that this didn't compile on Windows, does this need a private counterpart? The second one is a comment inline. Thanks!

**Comment by townsend2010** [comment]
Hey @luis4a0!

Thanks for the review!

> The first one is that this didn't compile on Windows, does this need a private counterpart?

Ah, right, the branch I'm working on for SMB mounts does implement this for Windows, but this should have a private counterpart that adds a stub for this.  I'll fix that.

File: src/client/cli/cmd/mount.cpp (hunk 1 above)
Comment (townsend2010) [reply src/client/cli/cmd/mount.cpp:108]: Well, it's not *quite* verbatim, the request type is different in each case.  I did give it some thought on factoring this out, but then we'd have to template the function and that gets messing with a free function, so then I'd need to include it in a class and implement it in the header.  I agree it seems quite redundant.  I'll give it a bit more thought.

Comment (luis4a0) [reply src/client/cli/cmd/mount.cpp:108]: Oh, right, I didn't note that. I'm OK with repeating few lines of code in that case.

Comment (townsend2010) [reply src/client/cli/cmd/mount.cpp]: Ok, refactored this some more.

**Review by luis4a0 — APPROVED** [review]
Yeah, looks great! Thanks!

**Pull request merged**